"duplicate class" compiler errors

As first reported at #103, Enunciate sometimes fails with "duplicate class" compiler errors that looks something like this:

```
[INFO] --- enunciate-maven-plugin:2.0.0-M.1:docs (default-cli) @ NewCRM ---
(... downloads lots of -source.jar ...)
[INFO] Reflections took 1760 ms to scan 94 urls, producing 9139 keys and 9139 values 
/home/garagoth/.m2/repository/net/sf/ehcache/ehcache/2.10.0/ehcache-2.10.0-sources.jar(net/sf/ehcache/statistics/extended/ExtendedStatisticsImpl.java):39: error: cannot access Query
import org.terracotta.context.query.Query;
                                   ^
  bad source file: /home/garagoth/.m2/repository/net/sf/ehcache/ehcache/2.10.0/ehcache-2.10.0-sources.jar(org/terracotta/context/query/Query.java)
    file does not contain class org.terracotta.context.query.Query
    Please remove or make sure it appears in the correct subdirectory of the sourcepath.
/home/garagoth/.m2/repository/net/sf/ehcache/ehcache/2.10.0/ehcache-2.10.0-sources.jar(net/sf/ehcache/util/concurrent/LongAdder.java):40: error: duplicate class: net.sf.ehcache.util.concurrent.LongAdder
public class LongAdder extends Striped64 implements Serializable {
       ^
/home/garagoth/.m2/repository/net/sf/ehcache/ehcache/2.10.0/ehcache-2.10.0-sources.jar(org/terracotta/statistics/derived/EventParameterSimpleMovingAverage.java):27: error: cannot access LongAdder
import org.terracotta.statistics.jsr166e.LongAdder;
                                        ^
  bad source file: /home/garagoth/.m2/repository/net/sf/ehcache/ehcache/2.10.0/ehcache-2.10.0-sources.jar(org/terracotta/statistics/jsr166e/LongAdder.java)
    file does not contain class org.terracotta.statistics.jsr166e.LongAdder
    Please remove or make sure it appears in the correct subdirectory of the sourcepath.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.1:docs (default-cli) on project NewCRM: Execution default-cli of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.1:docs failed: Enunciate compile failed. -> [Help 1]
```

I get this error sporadically. Not sure if it's related to this, but it looks very similar. I don't get a "duplicate class" message, but the rest looks similar.

```
10:28:21 [INFO] Building <artifactName> <artifactVersion>
10:28:21 [INFO] ------------------------------------------------------------------------
10:28:22 [INFO] 
10:28:22 [INFO] --- enunciate-maven-plugin:2.2.0:docs (default-cli) @ <artifactId> ---
10:28:22 [INFO] [ENUNCIATE] Using enunciate configuration at /home/jenkins/workspace/<jenkinsJob>/<groupId>/<artifactId>/enunciate.xml
10:28:23 [INFO] Reflections took 692 ms to scan 49 urls, producing 830 keys and 830 values 
10:28:25 [WARNING] [ENUNCIATE] [javac] [ERROR] com.webcohesion.enunciate.Enunciate$URLFileObject[file:/home/jenkins/.m2/repository/<projectPath>/<anotherArtifactId>/<artifactVersion>/<anotherArtifactId>-<artifactVersion>-sources.jar!/<projectPath>/domain/search/SearchField.java]:5:36 /home/jenkins/.m2/repository/<projectPath>/<anotherArtifactId>/<artifactVersion>/<anotherArtifactId>-<artifactVersion>-sources.jar!/<projectPath>/domain/search/SearchField.java:5: error: cannot access <anotherPackageName>.common.Presentation
10:28:25 [WARNING] [ENUNCIATE] [javac] /home/jenkins/.m2/repository/<projectPath>/<anotherArtifactId>/<artifactVersion>/<anotherArtifactId>-<artifactVersion>-sources.jar!/<projectPath>/domain/search/SearchField.java:5: error: cannot access <anotherPackageName>.common.Presentation
10:28:25 [WARNING] [ENUNCIATE] [javac] import <anotherPackageName>.common.Presentation;
10:28:25 [WARNING] [ENUNCIATE] [javac]                                    ^
10:28:25 [WARNING] [ENUNCIATE] [javac]   bad source file: /home/jenkins/.m2/repository/<projectPath>/<anotherArtifactId>/<artifactVersion>/<anotherArtifactId>-<artifactVersion>-sources.jar(<projectPath>/domain/common/Presentation.java)
10:28:25 [WARNING] [ENUNCIATE] [javac]     file does not contain class <anotherPackageName>.common.Presentation
10:28:25 [WARNING] [ENUNCIATE] [javac]     Please remove or make sure it appears in the correct subdirectory of the sourcepath.
10:28:25 [JENKINS] Archiving disabled
10:28:25 [JENKINS] Archiving disabled
10:28:25 [INFO] ------------------------------------------------------------------------
10:28:25 [INFO] BUILD FAILURE
10:28:25 [INFO] ------------------------------------------------------------------------
10:28:25 [INFO] Total time: 4.936s
10:28:25 [INFO] Finished at: Wed Feb 10 10:28:25 CET 2016
10:28:25 [INFO] Final Memory: 28M/389M
10:28:25 [INFO] ------------------------------------------------------------------------
10:28:25 [ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.2.0:docs (default-cli) on project <artifactId>: Execution default-cli of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.2.0:docs failed: Enunciate compile failed. -> [Help 1]
10:28:25 [ERROR] 
10:28:25 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
10:28:25 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
10:28:25 [ERROR] 
10:28:25 [ERROR] For more information about the errors and possible solutions, please read the following articles:
10:28:25 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
10:28:25 channel stopped
10:28:26 Finished: FAILURE
```

--- a/core/src/main/java/com/webcohesion/enunciate/Enunciate.java
+++ b/core/src/main/java/com/webcohesion/enunciate/Enunciate.java
@@ -585,7 +585,7 @@ public class Enunciate implements Runnable {
       getLogger().debug("Detected API Types: %s", new EnunciateLogger.ListWriter(includedTypes));
 
       //gather all the java source files.
-      List<URL> sourceFiles = getSourceFileURLs();
+      Set<URL> sourceFiles = getSourceFileURLs();
       URLClassLoader apiClassLoader = new URLClassLoader(scanpath.toArray(new URL[0]));
       for (String javaFile : scannedSourceFiles) {
 
@@ -771,8 +771,8 @@ public class Enunciate implements Runnable {
     return builder.toString();
   }
 
-  protected List<URL> getSourceFileURLs() {
-    List<URL> sourceFiles = new ArrayList<>(this.sourceFiles.size());
+  protected Set<URL> getSourceFileURLs() {
+    Set<URL> sourceFiles = new HashSet<>(this.sourceFiles.size());
     for (File sourceFile : this.sourceFiles) {
       try {
         sourceFiles.add(sourceFile.toURI().toURL());

